include/yaramod/types/expressions.h

```diff
@@ -1355,7 +1333,7 @@ class FunctionCallExpression : public IdExpression
 public:
 	template <typename ExpPtr, typename ExpPtrVector>
 	FunctionCallExpression(ExpPtr&& func, TokenIt left_bracket, ExpPtrVector&& args, TokenIt right_bracket)
-		: IdExpression()
+		: IdExpression(std::static_pointer_cast<const IdExpression>(func)->getSymbol())
```

Comment:
Do we mind this cast here? I wanted to preserve the ExpPtr template for perfect forwarding...

Reply:
It should be fine if used properly. And since it only should be created through builder then I consider it completely fine.

Reply:
Perfect. It is used in the parser too, but carefully. It is only passed an expression that is actually IdExpression or derived class. (The "identifier" grammar rule always creates some IdExpression...)